### PR TITLE
feat(operator) : Hello Eigen world ( 🦀 )

### PR DESCRIPTION
**Pull request by supernovahs**
Acceptance criteria:
- [x] Readme
- [x] test for the operator crate
- [x] test for the task crate

----
ECDSA Stake Registry is not yet deployed on Holesky officially. Hence I have commented out the code for `registerOperatorWithSignature` for now . If anyone wants to do it , they can uncomment the line where we call the function.

Also, the `registerOperator ` call in DelegationManager is also commented for example purposes , as our private key is already registered once there.

### CODE REVIEW

**Comment by supernovahs** [comment]
@NimaVaziri waiting for your  review  on the readme. Will then take this  over the finish line

**Review by Gajesh2007 — COMMENTED** [review]
can you nest the operator code like this:

`operator/rust/`

**Comment by Gajesh2007** [comment]
can you add README to run the scripts?

**Review by supernovahs — COMMENTED** [review]
@Gajesh2007 already added here for anvil 

For holesky , no scripts are required

**Comment by NimaVaziri** [comment]
@supernovahs let's add some tests to the crates here as well

**Review by Gajesh2007 — COMMENTED** [review]
can u keep all the json abis in one place and import it for typescript and rust?

File: README.md [hunk 1]
```diff
@@ -32,11 +32,12 @@ Where additional sophistication with AVSs come into the picture:
 2. [Foundry](https://getfoundry.sh/)
 3. [Docker](https://www.docker.com/get-started/)
 
+
 Following NodeJS packages:
 1. tcs
 1. ethers
 
-### Steps
+### Go lang Steps
```
[the remark this answers is not in the document]
Comment (supernovahs) [reply README.md:40]: lol  😆

File: .env [hunk 2]
```diff
@@ -3,4 +3,12 @@ RPC_URL=http://127.0.0.1:8545
 CONTRACT_ADDRESS=0x84eA74d481Ee0A5332c457a4d796187F6Ba67fEB
 DELEGATION_MANAGER_ADDRESS=0xCf7Ed3AccA5a467e9e704C703E8D87F634fB0Fc9
 STAKE_REGISTRY_ADDRESS=0x9E545E3C0baAB3E08CdfD552C960A1050f373042
-AVS_DIRECTORY_ADDRESS=0x5FC8d32690cc91D4c39d9d3abcBD16989F875707
\ No newline at end of file
+AVS_DIRECTORY_ADDRESS=0x5FC8d32690cc91D4c39d9d3abcBD16989F875707
+RUST_PRIVATE_KEY=0x70cf3589410a4471117581a3ed6aed567d180ad96b51d97bf2744a9c74fe55d0
+RUST_RPC_URL=https://1rpc.io/holesky
+RUST_CONTRACT_ADDRESS=0x3361953F4a9628672dCBcDb29e91735fb1985390
```
[the remark this answers is not in the document]
Comment (supernovahs) [reply .env:9]: yup

[the remark this answers is not in the document]
Comment (supernovahs) [reply .env:9]: Should I rename like HOLESKY_PRIVATE_KEY?

[the remark this answers is not in the document]
Comment (supernovahs) [reply .env]: I changed the name to holesky_

File: Cargo.toml
Comment (Gajesh2007) [inline Cargo.toml:1]: remove authors from here

**Comment by supernovahs** [comment]
I don't think we can use same json abi files for both. Typescript one uses the `.ts` syntax and alloy(rust) one uses the `.json` syntax

File: Cargo.toml
Comment (supernovahs) [reply Cargo.toml:1]: Who should be the author then?
Or i completely remove the author tag?

File: README.md [hunk 3]
```diff
@@ -32,11 +32,12 @@ Where additional sophistication with AVSs come into the picture:
...
 
+
 Following NodeJS packages:
 1. tcs
 1. ethers
 
-### Steps
+### Typescript Steps
 
 1. Make sure Docker is running
```
Comment (NimaVaziri) [inline README.md:42]: Let's move this above as part of 3.1

**Review by NimaVaziri — COMMENTED** [review]
@supernovahs where in this are you using the rs sdk? Took a look but might have missed it

File: operator/rust/crates/operator/src/start_operator.rs [hunk 4]
```diff
@@ -0,0 +1,311 @@
...
+use reqwest::Url;
+use HelloWorldServiceManager::Task;
+
+use alloy_primitives::{eip191_hash_message, Address, FixedBytes, U256};
+use alloy_signer::Signer;
+use alloy_signer_wallet::LocalWallet;
+use eigen_client_elcontracts::{
+    reader::ELChainReader,
+    writer::{ELChainWriter, Operator},
+};
```
Comment (supernovahs) [inline operator/rust/crates/operator/src/start_operator.rs:21]: @NimaVaziri  this is where we are importing the sdk crate

**Review by supernovahs — COMMENTED** [review]
^ comment

File: README.md (hunk 3 above)
Comment (supernovahs) [reply README.md:44]: have pushed a change . let me know if this what you meant

**Review by NimaVaziri — APPROVED** [review]

**Comment by supernovahs** [comment]
@NimaVaziri  I can't merge this  myself.

**Pull request merged**